src/main/java/mateusz/michal/chat/Service/RegistrationService.java

```diff
@@ -162,20 +134,72 @@ private boolean isPasswordIncorrect(String password) {
         return errors;
     }
 
-    private List<MyError> validateRegistrationRequest(UserDTO userDTO){
-        List<MyError> errors = new ArrayList<>();
-        errors.addAll(validateUserName(userDTO.getName()));
-        errors.addAll(validateUserEmail(userDTO.getEmail()));
-        errors.addAll(validateUserPassword(userDTO.getPassword()));
-        return errors;
+    // methods that validate single elements of properties
+    private boolean isNameEmptyString(String name){
+        return name.equals("");
+    }
+
+    private boolean isNameIncorrect(String name){
+        return (name.startsWith(" ") || name.endsWith(" ") || hasNameSpecialCharacters(name));
+    }
+
+    private boolean isEmailEmptyString(String email){
+        return email.equals("");
+    }
+
+    private boolean isPassworEmptyString(String password){
+        return password.equals("");
+    }
+
+    private boolean hasNameSpecialCharacters(String name){
+        for(char c : name.toCharArray()){
+            if(!(Character.isLetterOrDigit(c))){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean isNameNull(String name){
+        return name == null;
     }
 
+    private boolean isEmailNull(String email){
+        return email == null;
+    }
+
+    private boolean isPasswordNull(String password){
+        return password == null;
+    }
+
+    private boolean isUserInDatabaseByName(String name){
+        Optional<User> user = Optional.ofNullable(userRepository.findByName(name));
+        return user.isPresent();
+    }
+
+    private boolean isUserInDatabaseByEmail(String email){
+        Optional<User> user = Optional.ofNullable(userRepository.findByEmail(email));
+        return user.isPresent();
+    }
+
+    private boolean isEmailIncorrect(String email){
+        Pattern pattern = Pattern.compile("[a-zA-Z]+[\\.a-zA-Z0-9]*@[a-zA-Z0-9]+\\.[a-z]{2,}[a-z]*");
+        return !pattern.matcher(email).matches();
+    }
+
+    private boolean isPasswordIncorrect(String password) {
```

Comment:
nit: myślę że warto jednak unikać zaprzeczeń w nazwach funkcji, potem ich użycie może być mylące, tzn. `isPasswordIncorrect` zwraca `false`, więc jest poprawne, czy nie... Niepotrzebnie trzeba się zastanawiać i stosować podwójne przeczenie :) Ale to tylko hint na przyszłość

Reply:
Idea była taka aby w metodzie która waliduje konkretne pole nie było już zaprzeczeń. 
Może faktycznie tak będzie czytelniej.

Reply:
To tylko nit - jeśli uważasz inaczej / nie chcesz tego poprawiać - możesz po prostu zamknąć dyskusję :)